Tests/ApolloCodegenTests/ApolloCodegenConfigurationCodableTests.swift

```diff
@@ -923,4 +923,49 @@ class ApolloCodegenConfigurationCodableTests: XCTestCase {
       ))
     )
   }
+
+  func test__decodeApolloCodegenConfiguration__withInvalidOptions() throws {
+    // given
+    let subject = """
+    {
+      "schemaName": "MySchema",
+      "input": {
+        "operationSearchPaths": ["/search/path/**/*.graphql"],
+        "schemaSearchPaths": ["/path/to/schema.graphqls"]
+      },
+      "output": {
+        "testMocks": {
+          "none": {}
+        },
+        "schemaTypes": {
+          "path": "./MySchema",
+          "moduleType": {
+            "swiftPackageManager": {}
+          }
+        },
+        "operations": {
+          "inSchemaModule": {}
+        },
+        "options": {
+          "selectionSetInitializers" : {
+            "operations": true,
+            "namedFragments": true,
+            "localCacheMutations" : true
+          },
+          "queryStringLiteralFormat": "multiline",
+          "schemaDocumentation": "include",
+          "apqs": "disabled",
+          "warningsOnDeprecatedUsage": "include"
+        }
+      }
+    }
+    """.asData
+
+    func decodeConfiguration(subject: Data) throws -> ApolloCodegenConfiguration {
+      try JSONDecoder().decode(ApolloCodegenConfiguration.self, from: subject)
+    }
+    XCTAssertThrowsError(try decodeConfiguration(subject: subject)) { error in
+      XCTAssert(error is DecodingError)
```

Comment:
Can we be more specific on testing that the error is the correct error?
```swift
guard case let DecodingError.typeMismatch(type, context) = error else {
  fail("Incorrect error type")
}

XCTAssert(type is $ExpectedType)
XCTAssertEqual(context.codingPath, $expectedPath)
```

Reply:
Addressed in [02f4072](https://github.com/apollographql/apollo-ios/pull/3125/commits/02f40727da07e760569353fb1d66142bda2d0e40)